.github/workflows/contract-test.yml

```diff
@@ -12,7 +12,7 @@ env:
   PACT_BROKER_BASE_URL: ${{ secrets.PACT_BROKER_BASE_URL }}
   PACT_BROKER_TOKEN: ${{ secrets.PACT_BROKER_TOKEN }}
   GITHUB_SHA: ${{ github.sha }}
-  GITHUB_BRANCH: ${{ github.ref_name }}
+  GITHUB_BRANCH: ${{ github.head_ref }}
```

Comment:
In GitHub Actions, when running workflows triggered by pull requests, the environment variables can differ from what you might expect. Specifically:

GITHUB_REF and GITHUB_REF_NAME: These refer to the merge ref (e.g., refs/pull/63/merge and 63/merge), which is a special ref used by GitHub Actions for testing the result of merging a pull request.

GITHUB_HEAD_REF: This contains the actual branch name of the pull request (e.g., fix/schema-test).